app/services/tag_service.rb

```diff
@@ -0,0 +1,17 @@
+class TagService
```

Comment:
Wydaje mi się że takie użycie service to spore nadużycie - jeśli robisz collecta po tagach w modelu to nie jest to usługa (user-oriented) tylko specyfikacja obiektu, więc jeśli już to należało by to przenieść do modułu z robić `include` (pod warunkiem że planujesz tego używać ponownie w innych modelach).